sdk/monitor/monitor-query/src/internal/modelConverters.ts

```diff
@@ -122,14 +126,45 @@ export function convertResponseForQueryBatch(
         return left - right;
       })
       ?.map((response: GeneratedBatchQueryResponse) => ({
-        id: response.id,
         status: response.status,
+        visualization: response.body?.render,
+        statistics: response.body?.statistics,
         // hoist fields from the sub-object 'body' to this level
         error: response.body?.error,
-        tables: response.body?.tables?.map(convertGeneratedTable)
-      }))
+        tables: response.body?.tables?.map((table) => convertGeneratedTable(table)),
+        logsQueryResultStatus: "Success"
+      })),
+    batchResultStatus: "AllSucceeded"
   };
-
+  // compute status for failed or succeed or partial results
+  if (newResponse.results) {
```

Comment:
Question: is this response successful?

```ts
  newResponse = {
    results: []
  }
```

Just curious. The logic here would say "yes" but I am not familiar enough with the service to know

Reply:
The service can return an empty result set unfortunately. We are just following the swagger.

Reply:
should i update the result status there to something?